ballerina/schema-client.bal

```diff
@@ -0,0 +1,319 @@
+// Copyright (c) 2022 WSO2 LLC. (https://www.wso2.com) All Rights Reserved.
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/sql;
+
+public type TableDefinition record {
+    *sql:TableDefinition;
+    sql:CheckConstraint[] checkConstraints?;
+};
+
+# Represents an SQL metadata client.
+isolated client class SchemaClient {
+    private final Client dbClient;
+    private final string database;
+
+    # Initializes the SchemaClient object
+    #
+    # + host - The name of where the mysql server is hosted (ex: localhost)
+    # + user - The username to access the database
+    # + password - The password to access the database
+    # + database - The name of the database to be accessed
+    # + port - The port the database will be accessed on
+    # + options - MySQL database options
+    # + connectionPool - The `sql:ConnectionPool` to be used for the connection. If there is no
+    #                    `connectionPool` provided, the global connection pool (shared by all clients) will be used                  
+    # + return - An `sql:Error` or `()`
+    public function init(string host, string user, string password, string database, int port, 
+            Options? options = (), sql:ConnectionPool? connectionPool = ()) returns sql:Error? {
+        self.database = database;
+        self.dbClient = check new (host, user, password, database, port, options, connectionPool);
```

Comment:
Do we need to provide the `database` when initializing the db client?

Reply:
To run most of the MySQL queries I use the database as one of the while conditions. It would be more convenient to include the database as a parameter in the init method rather than requiring the user to specify it in each function.